test/crud-with-options.test.js

```diff
@@ -28,7 +28,36 @@ describe('crud-with-options', function() {
     options = {};
     filter = {fields: ['name', 'id']};
 
-    db.automigrate(['User'], done);
+    function maiLSetter(value) {
+      if (value !== 'NotAAchance') {
+        throw new Error('InProperValue');
+      }
+      this.__data.email = value;
+    }
+    FailingUser = db.define('FailingUser', {
+      id: {type: Number, id: true},
+      seq: {type: Number, index: true},
+      name: {type: String, index: true},
+      email: {type: String, index: true},
+    });
+    FailingUser.setter.email = maiLSetter;
+    db.automigrate(['User', 'FailingUser'], done);
+  });
+
+  describe('create', () => {
+    it('should return rejected promise when model initialization failed', async () => {
+      await FailingUser.create({id: 1, seq: 1, name: 'Sad Fail', email: 'some@some.com'}).should
+        .be.rejectedWith('InProperValue');
+    });
+  });
+
+  describe('replaceById', () => {
+    it('should return rejected promise when model initialization failed', async () => {
+      const firstNonFailedUser = FailingUser.create({id: 1, seq: 1, name: 'Sad Fail'}); // no email no fail
+      await FailingUser.replaceById(firstNonFailedUser.id, {
+        seq: 1, name: 'Sad Fail', email: 'some@some.com',
+      }).should.be.rejectedWith('InProperValue');
+    });
```

Comment:
This is not the right place where to add your new tests. As the file name `crud-with-options.test.js` suggests, these tests are covering invocation of DAO methods with `options` argument. AFAICT, your new tests don't care about `options`.

Please move your new tests to [test/manipulation.test.js](https://github.com/strongloop/loopback-datasource-juggler/blob/master/test/manipulation.test.js).

It would be great if you could find a way how to avoid creating a new model that will be shard with all other tests in the file. 

For example, you can keep using the shared `Person` model, but add a new property that throws when it's set to a non-empty value. The important part is to give it a descriptive name, e.g. `throwingSetter`.

```ts
Person = db.define('Person', {
  // ...
  throwingSetter: {type: String},
}, {forceId: true, strict: true});

Person.setter.throwingSetter = function(value) {
  if (!value) return; // no-op
  throw new Error('Intentional error triggered from a property setter.');
};
```

This will make it easy in the future to write new tests related to setters that are throwing.